api/scpca_portal/models/project.py

```diff
@@ -508,19 +513,11 @@ def get_multiplexed_with_mapping(self, multiplexed_libraries_metadata: List[Dict
         and the values being all ids that this sample was multiplexed with.
         """
         multiplexed_with_mapping = {}
-        for library_metadata in multiplexed_libraries_metadata:
-            multiplexed_library_sample_ids = library_metadata["demux_samples"]
-            for multiplexed_sample_id in multiplexed_library_sample_ids:
-                # Remove sample ID from a mapping as sample cannot be
-                # multiplexed with itself.
-                multiplexed_library_sample_ids_copy = set(multiplexed_library_sample_ids)
-                multiplexed_library_sample_ids_copy.discard(multiplexed_sample_id)
-
-                # Populate multiplexed sample mapping.
-                if multiplexed_sample_id not in multiplexed_with_mapping:
-                    multiplexed_with_mapping[multiplexed_sample_id] = set()
-                multiplexed_with_mapping[multiplexed_sample_id].update(
-                    multiplexed_library_sample_ids_copy
+        for multiplexed_dir in sorted(Path(self.input_data_path).rglob("*,*/")):
+            multiplexed_ids = multiplexed_dir.name.split(",")
+            for multiplexed_id in multiplexed_ids:
+                multiplexed_with_mapping[multiplexed_id] = set(
+                    id for id in multiplexed_ids if id is not multiplexed_id
                 )
```

Comment:
```suggestion
            multiplexed_ids = set(multiplexed_dir.name.split(","))
            for id in multiplexed_ids:
                multiplexed_with_mapping[id] = multiplexed_ids.difference(id)                )
```

I think that doing the substring check made this a bit harder to read. what do you think about this approach instead? - also need to remove the trailing `)`

Reply:
This is great! I'll add it.